rllib/algorithms/algorithm_config.py

```diff
@@ -568,6 +570,13 @@ def update_from_dict(
         """
         eval_call = {}
 
+        # We deal with this special key before all others because it may influence
+        # stuff like "exploration_config".
+        # Namely, we want to re-instantiate the exploration config this config had
+        # inside `self.rl_module()` before potentially overwriting it in the following.
+        if "_enable_rl_module_api" in config_dict:
+            self.rl_module(_enable_rl_module_api=config_dict["_enable_rl_module_api"])
```

Comment:
Further down, there is special handling of `exploration_config`.
With the rest of the logic in this PR, we must properly enable rl module API before dealing with the exploration config. Otherwise we might override the exploration config that is being merged.